include/openssl/thread.h

```diff
@@ -59,6 +59,10 @@
 
 #include <sys/types.h>
 
+#if defined(__OpenBSD__)
```

Comment:
Nit: This should be moved to `include/openssl/base.h` and defined as `OPENSSL_OPENBSD`, which should be used here instead. This is how it's done for other operating systems (e.g. FreeBSD).

Reply:
Updated the diff. Please let me know how close/far I am to what you were expecting.